src/django_components/dependencies.py

```diff
@@ -492,16 +493,25 @@ def get_component_media(comp_cls_hash: str) -> Media:
     loaded_css_urls = sorted(
         [
             *loaded_component_css_urls,
-            # NOTE: Unlike JS, the initial CSS is loaded outside of the dependency
-            # manager, and only marked as loaded, to avoid a flash of unstyled content.
-            *to_load_css_urls,
+            # NOTE: When rendering a document, the initial CSS is inserted directly into the HTML
+            # to avoid a flash of unstyled content. In the dependency manager, we only mark those
+            # scripts as loaded.
+            *(to_load_css_urls if type == "document" else []),
+        ]
+    )
+    loaded_js_urls = sorted(
+        [
+            *loaded_component_js_urls,
+            # NOTE: When rendering a document, the initial JS is inserted directly into the HTML
+            # so the scripts are executed at proper order. In the dependency manager, we only mark those
+            # scripts as loaded.
+            *(to_load_js_urls if type == "document" else []),
```

Comment:
What this bit of changes does is that before, we've sent the JS scripts from `Media.js` to be loaded via our client-side `loadJs()` function (previously called `loadScript()`.

But since now we insert them as regular `<script>` tags into the HTML, we just let the dependency manager know that these scripts have been loaded by calling `markScriptLoaded()` with the URL of the resource.